### DISCUSSION TITLE
Having issues pushing my codes to GitHub… and I need help..

### DISCUSSION THREAD

**Discussion by Rexedoziem** [community/community#132869] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Feature Area

Issues

### Body

HP@DESKTOP-IPU1ME1 MINGW64 ~/Desktop/RESTAURANT_COMPONENTS (master)
$ git push -u origin master
Uploading LFS objects: 100% (6/6), 1.0 GB | 0 B/s, done.
Enumerating objects: 52215, done.
Counting objects: 100% (52215/52215), done.
Delta compression using up to 4 threads
Compressing objects: 100% (41811/41811), done.
Writing objects: 100% (52213/52213), 620.63 MiB | 11.06 MiB/s, done.
Total 52213 (delta 9933), reused 52182 (delta 9922), pack-reused 0
error: RPC failed; curl 6 Could not resolve host: github.com
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly
Everything up-to-date

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by knarrff** [comment]
> Could not resolve host: github.com

This means git cannot connect to github, it cannot even resolve the name. Do you have any kind of special network setup? Do you maybe need to use a proxy to connect to the outside? Do you use a VPN? Or did you ever tell git to use a proxy? You could check that by looking for 'proxy' in the output of ```git config --global -l``` and ```git config --local -l```.

**Reply by Rexedoziem** [reply]
$ git config --global -l
user.email=edoziemenyinnaya244@gmail.com
user.name=Rexedoziem
credential.helperselector.selected=store
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
filter.lfs.clean=git-lfs clean -- %f
http.postbuffer=[1048576000](tel:1048576000)


$ git config --local -l
core.repositoryformatversion=0
core.filemode=false
core.bare=false
core.logallrefupdates=true
core.symlinks=false
core.ignorecase=true
remote.origin.url=https://github.com/Rexedoziem/GNN-Powered-Restaurant-Recommender.git
remote.origin.fetch=+refs/heads/:refs/remotes/origin/
lfs.repositoryformatversion=0
lfs.https://github.com/Rexedoziem/GNN-Powered-Restaurant-Recommender.git/info/lfs.access=basic

This is the output of both….

**Reply by knarrff** [reply]
This does not have a proxy set. Do you need one with your internet connection? What happens when you, on the command line, type ```host github.com```? What about ```git pull``` inside your repo: that should in theory give the same error. If this doesnt, but push continues to give you an error, post the output of ```git remote -v```. Usually, it shows at least one entry for (fetch) and one for (push) and the URL for both usually is the same. See if it is.

**Reply by Rexedoziem** [reply]
So how do I go about it?

**Reply by knarrff** [reply]
In the same place you typed in ```git config --local -l``` earlier, type the following commands and copy&paste the output here:

- ```host github.com```
- ```git pull``` (for this command: be inside the clone of your repository, according you your question that might be ```~/Desktop/RESTAURANT_COMPONENTS```)
- ```git remote -v``` (for this command: be inside the clone of your repository, according you your question that might be ```~/Desktop/RESTAURANT_COMPONENTS```)

**Reply by Rexedoziem** [reply]
host command not found so I used ping to GitHub…
$ ping [github.com](http://github.com/)

Pinging [github.com](http://github.com/) [140.82.121.4] with 32 bytes of data:
Reply from [140.82.121.4](http://140.82.121.4/): bytes=32 time=282ms TTL=45
Reply from [140.82.121.4](http://140.82.121.4/): bytes=32 time=145ms TTL=45
Reply from [140.82.121.4](http://140.82.121.4/): bytes=32 time=231ms TTL=45
Reply from [140.82.121.4](http://140.82.121.4/): bytes=32 time=140ms TTL=45

Ping statistics for [140.82.121.4](http://140.82.121.4/):
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 140ms, Maximum = 282ms, Average = 199ms

$ git remote -v
origin  https://github.com/Rexedoziem/GNN-Powered-Restaurant-Recommender.git (fetch)
origin  https://github.com/Rexedoziem/GNN-Powered-Restaurant-Recommender.git (push)


$ git pull
Already up to date.

**Reply by knarrff** [reply]
Good, this at least rules out general connection problems.

I noticed this in your log just now: "Writing objects: 100% (52213/52213), 620.63 MiB "
This is a lot of data for one commit. It's nothing git couldn't handle, but maybe the internet connection can't, especially if it is slow (is it?): things could run into timeouts or overflow buffer sizes.

Try (within your repo):

```git config --global http.postBuffer 524288000``` and re-run the push. If this does not work, try increasing the number to 1048576000 and retry. If this does not work, also try:
```git config --local http.version HTTP/1.1``` and again re-run the push.

**Reply by Rexedoziem** [reply]
I’ll try that out now…..

This is what I got from GitHub:
Git LFS has been disabled on your personal account Rexedoziem because you’ve exceeded your data plan by at least 150%. Please purchase additional data packs to cover your bandwidth and storage usage:

 https://github.com/account/billing/data/upgrade

Current usage as of 20 Jul 2024 09:14PM UTC:

 Bandwidth: 0.0 GB / 1 GB (0%)
 Storage: 1.62 GB / 1 GB (162%)

**Reply by knarrff** [reply]
I see. Maybe this is already the answer: you may be 'out of space'.

**Reply by Rexedoziem** [reply]
So what can be done about it?

**Reply by Rexedoziem** [reply]
Or can I create a new repo and start the same process?

**Reply by knarrff** [reply]
github gives you 1TB free space. If you want more, you can purchase more. See the URL above: https://github.com/account/billing/data/upgrade

... or you figure out how to use less, if that is possible.

**Reply by knarrff** [reply]
> Or can I create a new repo and start the same process?

I believe the data limits are per user, not per repo.

**Reply by Rexedoziem** [reply]
But can I remove those large files from GitHub and just upload it like that?

**Reply by knarrff** [reply]
Try the steps here: https://docs.github.com/en/repositories/working-with-files/managing-large-files/removing-files-from-git-large-file-storage. Quoting from there the parts that may be most relevant here:

"_After you remove files from Git LFS, the Git LFS objects still exist on the remote storage and will continue to count toward your Git LFS storage quota. To remove Git LFS objects from a repository, delete and recreate the repository._"

**Reply by Rexedoziem** [reply]
But can I send the repo to GitHub without the  files from Git LFS?

**Reply by Rexedoziem** [reply]
If I can, get I get the commands for it?. Thanks

**Comment by KhujamovCodes** [comment]
Response #1

**Comment by thenikhilranjan** [comment]
R#1

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Rexedoziem ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Comment by 1543siddhant** [comment]
It looks like there was an issue with resolving the host "github.com" during your git push operation. This typically happens due to network issues or DNS resolution problems on your local machine.

Here are a few steps you can try to resolve this issue:

Check Network Connection: Ensure that your internet connection is stable and working properly.

DNS Resolution: Sometimes DNS cache issues can cause problems. Try flushing your DNS cache:

On Windows, open Command Prompt and run: ipconfig /flushdns
On macOS/Linux, open Terminal and run: sudo dscacheutil -flushcache; sudo killall -HUP mDNSResponder
Retry Push: After checking your network and clearing DNS cache, retry the git push command.

If the issue persists, it might be related to temporary server problems on GitHub's end. In such cases, waiting for a while and trying again later can also resolve the problem.

Hope this helps!
